### ISSUE TITLE
How to fix pylint E1134: Non-mapping value X is used in a mapping context (not-a-mapping)

### ISSUE THREAD

**Issue by 1Mark** [PyCQA/pylint#5810]
### Question

How to fix pylint E1134: Non-mapping value X is used in a mapping context (not-a-mapping)

```python
from dataclasses import dataclass
@dataclass
class One:
    f_one: str
    f_two: str

@dataclass
class Two:
    f_three: str
    f_four: One
    def __post_init__(self):
        """Initialise nested dataclass fields from a dictionary"""
        self.f_four = One(**self.f_four)

data = {'f_three': 'three', 'f_four': {'f_one': 'one', 'f_two': 'two'}}

print(Two(**data)) # this line is where it complaints
```

### Documentation for future user

Maybe an example in the documentation since this isnt that obvious in my opinion and I was using 2.9.3 and upgraded to 2.12.2 and this suddenly became an error.

### Additional context

For more details please see 

https://stackoverflow.com/questions/71118967/how-to-fix-pylint-e1134-non-mapping-value-x-is-used-in-a-mapping-context-not-a

**Comment by Pierre-Sassoulas** [comment]
Sorry for the time it took to answer. It look like it was a false positive that was fixed since. So I would say upgrade to latest pylint or disable the check but there's nothing to fix here.

**Comment by Pierre-Sassoulas** [comment]
My bad it still reproduce but it's a false positive.
